File: Learn_Basic/src/learn_py/learn_py/helloworld.py
import rclpy
from rclpy.node import Node
import time

def main(args=None):
    rclpy.init(args=args)
    node = Node("helloworld_test_node")
    
    while rclpy.ok():
        node.get_logger().info("hello world")
        time.sleep(0.5)
        
    node.destroy_node()
    rclpy.shutdown()

File: Learn_Basic/src/learn_cmake/src/helloworld.cpp
#include "rclcpp/rclcpp.hpp"
#include <chrono>
#include <thread>

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<rclcpp::Node>("hello_world_test_node");

    while (rclcpp::ok())
    {
        RCLCPP_INFO(node->get_logger(), "Hello, world!");
        std::this_thread::sleep_for(std::chrono::milliseconds(500));
    }

    rclcpp::shutdown();
    return 0;
}
